aten/src/ATen/native/cuda/EmbeddingBag.cu

```diff
@@ -264,6 +265,13 @@ _embedding_bag_cuda(const Tensor &weight, const Tensor &indices,
 
   int64_t numIndices = indices.size(0);
   int64_t numBags = offsets.size(0);
+  if (new_offsets) {
+    // Check https://github.com/pytorch/pytorch/issues/29019
+    // We plan to add one more element in offsets, which is equal to the size of
+    // indices. Currently for cuda devices, we still use the legacy
+    // implementation even this flag is enabled.
+    numBags -= 1;
```

Comment:
you want to add error checking here and in CPU branch that the offsets aren't empty when the new style is used